### PR TITLE
kubectl: Use fields from event series when computing describe events for a object

### LINKED ISSUE #1095: "kubectl describe <resource>" doesn't take into account the new fields of core.v1/event
When trying to implement sending Events from a controller, I decided to adhere to the new Event structure from core.v1. This has the notable changes that the `count` value and `lastTimestamp` are now deprecated in favor of `series.count` and `series.lastObservedTime`.

However, when using these fields, when invoking `kubectl describe deployment app1`, only `eventTime` is considered for the `Age` column.

When executing `kubectl get event <event-name>`, the `LAST SEEN` column is giving the right value and properly updates.

Expected behavior:

When executing `kubectl describe <resource> <name>`, the correct `lastObservedTime` and `count` should be shown.

### PR DESCRIPTION
**Pull request by harjas27**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
When using `kubectl describe <resource> <name>` , the events returned for the resource do not use the `EventSeries.Count` and `EventSeries.LastObservedTime` 
This PR adds the logic to compute the age of the event using these fields

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1095

#### Does this PR introduce a user-facing change?

```release-note
Changed kubectl describe to compute Age of an event using the count and lastObservedTime fields available in the event series 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go
Comment (KnVerey) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:4148]: Why `e.Series.Count+1`? Strictly speaking [`e.Series`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L5545) should be nil when only one event has occurred:

```go
	// Data about the Event series this event represents or nil if it's a singleton Event.
	// +optional
	Series *EventSeries `json:"series,omitempty" protobuf:"bytes,11,opt,name=series"`
```

I can see the argument that since we have an event, `e.Series.Count` should logically be at least 1. However, if the field does contain a number, I don't think we should increment it.

Take a look at this example [from the KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/383-new-event-api-ga-graduation/README.md#short-examples):

```yaml
# After second occurrence, Event looks like:

{
  regarding: A,
  action: B,
  reportingController: C,
  ...,
  series: {count: 2},
}
```

Comment (harjas27) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:4148]: I had added the `+1` following a comment in the code:
https://github.com/kubernetes/kubernetes/blob/7a0638da76cb9843def65708b661d2c6aa58ed5a/pkg/printers/internalversion/printers.go#L1789-L1793

Also referred [this](https://github.com/kubernetes/client-go/blob/a31b18a6ac98bbff0dd1055707dfde1b0a98ff68/tools/events/event_broadcaster.go#L183-L186). When a series is created for the first time, its count is set to 1 and for the series to be created, there needs to be a singleton event created.

Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:4148]: @KnVerey good catch, it looks like at the time the printer PR merged the validation wasn't fully complete. If you check the current code https://github.com/kubernetes/kubernetes/blob/758ad0790ceae9e7be0554637c0fd623ff02e45f/pkg/apis/core/validation/events.go#L110-L117 you'll notice that the minimum required value for counter is 2 and not one like from the above script. Which means that the series represent values exactly like describe in the KEP and not in client-go. So we need to drop `+1` from this PR and also from the printers.

**Review by soltysh — APPROVED** [review]
/approve
/hold

You still need to address @KnVerey comment dropping `+1` from counter and fix the release notes. @KnVerey has the final approval on the PR

**Comment by KnVerey** [comment]
@harjas27 please update the test assertion and add a release note to your PR description

**Comment by KnVerey** [comment]
/label tide/merge-method-squash
/unhold
/lgtm
/approve

Thank you @harjas27 ! Would you be interested in contributing the fix to remove the extra +1 from [the printer code you found](https://github.com/kubernetes/kubernetes/blob/7a0638da76cb9843def65708b661d2c6aa58ed5a/pkg/printers/internalversion/printers.go#L1789-L1793) in a follow up PR?

**Pull request merged**
